### PR TITLE
Composition

### PR DESCRIPTION
**Pull request by StarMindz**
I added three decorator classes
- BaseDecorator 
- capitalizeDecorator and 
- trimmerDecorator

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```
Comment (rloterh) [inline base_decorator.rb]: - [ ] OPTIONAL
It's a best practice to mark `nameable` with `attr_accessor`, to give an indication that we intend to both _read_ and write this variable. Let's endeavor to improve upon the same.

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    length = @nameable.correct_name.length
+    if length <= 10
+      @nameable.correct_name
+    else
+      @nameable.correct_name[0..10]
```
Comment (rloterh) [inline trimmer_decorator.rb]: - [ ] The `corect_name` function in the _Trimmer Decorator_ should return the given name with a maximum of **10** characters. If it's longer it should trim the word. Your  `corect_name` function is however, returning characters more than 10. Kindly ensure to improve upon the same.

**Characters more than 10 returned:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Stanley, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦 🚦  🏃 

## Highlights 👍  
- Nice job implementing Decorator pattern so far!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly add an appropriate instruction on how to run your app, under the _Getting Started_ section.
- Kindly endeavor to remove the guide note below the MIT license.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @StarMindz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**